client/state/data-layer/wpcom/sites/jitm/index.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { isJetpackSite } from 'state/sites/selectors';
+import { JITM_SET, SECTION_SET, SELECTED_SITE_SET } from 'state/action-types';
+import config from 'config';
+import { makeParser } from 'state/data-layer/wpcom-http/utils';
+import schema from './schema.json';
+
+/**
+ * Poor man's process manager
+ * @type {{
+ * hasInitializedSites: boolean,
+ * hasInitializedSection: boolean,
+ * lastSection: null|string,
+ * lastSite: null|string,
+ * }}
+ */
+const process = {
+	hasInitializedSites: false,
+	hasInitializedSection: false,
+	lastSection: null,
+	lastSite: null,
+};
+
+const unescapeDecimalEntities = ( str ) => {
+	return str.replace( /&#(\d+);/g, ( match, entity ) => String.fromCharCode( entity ) );
+};
+
+const transformApiRequest = ( { data: jitms } ) => jitms.map( ( jitm ) =>
+	( {
+		message: unescapeDecimalEntities( jitm.content.message ),
+		description: unescapeDecimalEntities( jitm.content.description ),
+		featureClass: jitm.feature_class,
+		callToAction: unescapeDecimalEntities( jitm.CTA.message ),
+		id: jitm.id,
+	} )
+);
+
+const insertJITM = ( dispatch, siteId, messagePath, jitms ) =>
+	dispatch( {
+		type: JITM_SET,
+		keyedPath: messagePath + siteId,
+		jitms: jitms.map( jitm => (
+			{ ...jitm, lastUpdated: Date.now() }
+		) )
+	} );
+
+/**
+ * Processes the current state and determines if it should fire a jitm request
+ * @param {object} state The current state
+ * @param {function} dispatch The redux dispatch function
+ * @param {string} action The action being processed
+ * @return {undefined} Nothing
+ */
+export const fetchJITM = ( state, dispatch, action ) => {
+	if ( ! config.isEnabled( 'jitms' ) ) {
+		return;
+	}
+
+	if ( ! process.hasInitializedSites || ! process.hasInitializedSection ) {
+		return;
+	}
+
+	const currentSite = process.lastSite;
+
+	if ( ! isJetpackSite( state, currentSite ) ) {
+		return insertJITM( dispatch, currentSite, process.lastSection, [] );
+	}
```

Comment:
why are we injecting JITMs into state if the site isn't a Jetpack site? why not simply `return` without doing anything?